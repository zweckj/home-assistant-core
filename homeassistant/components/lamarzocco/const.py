"""Constants for the La Marzocco integration."""

from typing import Final

DOMAIN: Final = "lamarzocco"

POLLING_INTERVAL: Final = 30
UPDATE_DELAY: Final = 3

DEFAULT_CLIENT_ID: Final = "7_1xwei9rtkuckso44ks4o8s0c0oc4swowo00wgw0ogsok84kosg"
DEFAULT_CLIENT_SECRET: Final = "2mgjqpikbfuok8g4s44oo4gsw0ks44okk4kc4kkkko0c8soc8s"

DEFAULT_PORT_LOCAL: Final = 8081

MODEL_GS3_AV: Final = "GS3 AV"
MODEL_GS3_MP: Final = "GS3 MP"
MODEL_LM: Final = "Linea Mini"
MODEL_LMU: Final = "Micra"

BREW_ACTIVE: Final = "brew_active"

GLOBAL: Final = "global"
MON: Final = "mon"
TUE: Final = "tue"
WED: Final = "wed"
THU: Final = "thu"
FRI: Final = "fri"
SAT: Final = "sat"
SUN: Final = "sun"

DAYS: Final = [MON, TUE, WED, THU, FRI, SAT, SUN]

MACHINE_NAME: Final = "machine_name"
SERIAL_NUMBER: Final = "serial_number"
