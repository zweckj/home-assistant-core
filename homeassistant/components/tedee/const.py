"""Constants for the Tedee integration."""
from datetime import timedelta

DOMAIN = "tedee"
NAME = "Tedee"

SCAN_INTERVAL = timedelta(seconds=10)

CONF_LOCAL_ACCESS_TOKEN = "local_access_token"
CONF_USE_LOCAL_API = "use_local_api"
CONF_USE_CLOUD_API = "use_cloud_api"
CONF_CLOUD_BRIDGE_ID = "cloud_bridge_id"
