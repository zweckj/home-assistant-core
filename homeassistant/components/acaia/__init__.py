"""Initialize the Acaia component."""
from homeassistant.config_entries import ConfigEntry
from homeassistant.const import CONF_MAC, CONF_NAME, Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import config_validation as cv
from homeassistant.helpers.typing import ConfigType

from .acaiaclient import AcaiaClient
from .const import CONF_IS_NEW_STYLE_SCALE, DOMAIN
from .coordinator import AcaiaApiCoordinator

CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)

PLATFORMS = [Platform.BUTTON, Platform.SENSOR]


async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
    """Set up the Acaia component."""
    hass.data.setdefault(DOMAIN, {})
    return True


async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
    """Set up Acaia as config entry."""

    name = config_entry.data[CONF_NAME]
    mac = config_entry.data[CONF_MAC]
    is_new_style_scale = config_entry.data.get(CONF_IS_NEW_STYLE_SCALE, True)

    acaia_client = AcaiaClient(
        hass, mac=mac, name=name, is_new_style_scale=is_new_style_scale
    )

    hass.data[DOMAIN][config_entry.entry_id] = coordinator = AcaiaApiCoordinator(
        hass, acaia_client
    )

    await coordinator.async_config_entry_first_refresh()

    await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS)

    return True


async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
    """Unload a config entry."""

    unload_ok = await hass.config_entries.async_unload_platforms(
        config_entry, PLATFORMS
    )

    if unload_ok:
        hass.data[DOMAIN].pop(config_entry.entry_id)

    return unload_ok
