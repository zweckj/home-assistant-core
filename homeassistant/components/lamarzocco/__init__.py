"""The La Marzocco integration."""

import logging

from lmcloud.client_cloud import LaMarzoccoCloudClient
from lmcloud.const import MachineModel
from lmcloud.exceptions import AuthFail, RequestNotSuccessful

from homeassistant.config_entries import ConfigEntry
from homeassistant.const import (
    CONF_HOST,
    CONF_MODEL,
    CONF_NAME,
    CONF_PASSWORD,
    CONF_TOKEN,
    CONF_USERNAME,
    Platform,
)
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import ConfigEntryError

from .const import DOMAIN
from .coordinator import LaMarzoccoMachineUpdateCoordinator

PLATFORMS = [
    Platform.BINARY_SENSOR,
    Platform.BUTTON,
    Platform.NUMBER,
    Platform.SELECT,
    Platform.SENSOR,
    Platform.SWITCH,
    Platform.UPDATE,
]

_LOGGER = logging.getLogger(__name__)


async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Set up La Marzocco as config entry."""

    if (model := entry.data[CONF_MODEL]) in MachineModel:
        coordinator = LaMarzoccoMachineUpdateCoordinator(hass)
    else:
        raise ConfigEntryError(f"Invalid model {model}")

    await coordinator.async_config_entry_first_refresh()
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator

    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)

    return True


async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Unload a config entry."""

    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)

    if unload_ok:
        hass.data[DOMAIN].pop(entry.entry_id)

    return unload_ok


async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Migrate config entry."""
    if entry.version == 1:
        cloud_client = LaMarzoccoCloudClient(
            username=entry.data[CONF_USERNAME],
            password=entry.data[CONF_PASSWORD],
        )
        try:
            fleet = await cloud_client.get_customer_fleet()
        except (AuthFail, RequestNotSuccessful) as exc:
            raise ConfigEntryError("Migration failed") from exc

        assert entry.unique_id
        device = fleet[entry.unique_id]
        v2_data = {
            CONF_USERNAME: entry.data[CONF_USERNAME],
            CONF_PASSWORD: entry.data[CONF_PASSWORD],
            CONF_MODEL: device.model,
            CONF_NAME: device.name,
            CONF_TOKEN: device.communication_key,
        }

        if CONF_HOST in entry.data:
            v2_data[CONF_HOST] = entry.data[CONF_HOST]

        entry.version = 2
        hass.config_entries.async_update_entry(
            entry,
            data=v2_data,
        )
        _LOGGER.debug("Migrated La Marzocco config entry to version 2")
    return True
