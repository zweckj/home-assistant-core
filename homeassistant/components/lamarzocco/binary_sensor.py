"""Binary Sensor platform for La Marzocco espresso machines."""

from collections.abc import Callable
from dataclasses import dataclass

from homeassistant.components.binary_sensor import (
    BinarySensorDeviceClass,
    BinarySensorEntity,
    BinarySensorEntityDescription,
)
from homeassistant.config_entries import ConfigEntry
from homeassistant.core import HomeAssistant
from homeassistant.helpers.entity_platform import AddEntitiesCallback

from .const import BREW_ACTIVE, DOMAIN
from .entity import LaMarzoccoEntity, LaMarzoccoEntityDescription
from .lm_client import LaMarzoccoClient


@dataclass
class LaMarzoccoBinarySensorEntityDescriptionMixin:
    """Description of an La Marzocco Binary Sensor."""

    is_on_fn: Callable[[LaMarzoccoClient], bool]


@dataclass
class LaMarzoccoBinarySensorEntityDescription(
    LaMarzoccoEntityDescription,
    BinarySensorEntityDescription,
    LaMarzoccoBinarySensorEntityDescriptionMixin,
):
    """Description of an La Marzocco Binary Sensor."""


ENTITIES: tuple[LaMarzoccoBinarySensorEntityDescription, ...] = (
    LaMarzoccoBinarySensorEntityDescription(
        key="water_reservoir",
        translation_key="water_reservoir",
        device_class=BinarySensorDeviceClass.PROBLEM,
        icon="mdi:water-well",
        is_on_fn=lambda client: not client.current_status.get(
            "water_reservoir_contact"
        ),
        extra_attributes={},
    ),
    LaMarzoccoBinarySensorEntityDescription(
        key=BREW_ACTIVE,
        translation_key=BREW_ACTIVE,
        device_class=BinarySensorDeviceClass.RUNNING,
        icon="mdi:cup-water",
        is_on_fn=lambda client: bool(client.current_status.get(BREW_ACTIVE)),
        extra_attributes={},
    ),
)


async def async_setup_entry(
    hass: HomeAssistant,
    config_entry: ConfigEntry,
    async_add_entities: AddEntitiesCallback,
) -> None:
    """Set up binary sensor entities."""
    coordinator = hass.data[DOMAIN][config_entry.entry_id]

    async_add_entities(
        LaMarzoccoBinarySensorEntity(coordinator, hass, description)
        for description in ENTITIES
        if not description.extra_attributes
        or coordinator.lm.model_name in description.extra_attributes
    )


class LaMarzoccoBinarySensorEntity(LaMarzoccoEntity, BinarySensorEntity):
    """Binary Sensor representing espresso machine water reservoir status."""

    entity_description: LaMarzoccoBinarySensorEntityDescription

    @property
    def is_on(self) -> bool:
        """Return true if the binary sensor is on."""
        return self.entity_description.is_on_fn(self._lm_client)
