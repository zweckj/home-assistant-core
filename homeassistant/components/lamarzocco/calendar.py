"""Calendar platform for La Marzocco espresso machines."""
from collections.abc import Iterator
from datetime import datetime, timedelta

from homeassistant.components.calendar import CalendarEntity, CalendarEvent
from homeassistant.config_entries import ConfigEntry
from homeassistant.core import HomeAssistant
from homeassistant.helpers.entity_platform import AddEntitiesCallback

from .const import DOMAIN
from .entity import LaMarzoccoBaseEntity

CALENDAR_KEY = "auto_on_off_schedule"


async def async_setup_entry(
    hass: HomeAssistant,
    config_entry: ConfigEntry,
    async_add_entities: AddEntitiesCallback,
) -> None:
    """Set up switch entities and services."""

    coordinator = hass.data[DOMAIN][config_entry.entry_id]
    async_add_entities([LaMarzoccoCalendarEntity(coordinator, CALENDAR_KEY)])


class LaMarzoccoCalendarEntity(LaMarzoccoBaseEntity, CalendarEntity):
    """Class representing a La Marzocco Calendar."""

    _attr_translation_key = CALENDAR_KEY

    @property
    def event(self) -> CalendarEvent | None:
        """Return the next upcoming event."""
        # only need to check the next 6 days, because if we don't find anything there
        # then there is no event scheduled
        for date in self._asnyc_date_range(
            datetime.today(), datetime.today() + timedelta(days=6)
        ):
            scheduled = self._async_get_calendar_event(date)
            if scheduled:
                if scheduled.end > datetime.now():
                    continue
                return scheduled
        return None

    async def async_get_events(
        self,
        hass: HomeAssistant,
        start_date: datetime,
        end_date: datetime,
    ) -> list[CalendarEvent]:
        """Return calendar events within a datetime range."""

        events: list[CalendarEvent] = []

        for date in self._asnyc_date_range(start_date, end_date):
            scheduled = self._async_get_calendar_event(date)
            if scheduled:
                events.append(scheduled)
        return events

    def _asnyc_date_range(
        self, start_date: datetime, end_date: datetime
    ) -> Iterator[datetime]:
        current_date = start_date
        while current_date <= end_date:
            yield current_date
            current_date += timedelta(days=1)

    def _async_get_calendar_event(self, date: datetime) -> CalendarEvent | None:
        """Return calendar event for a given weekday."""

        # check first if auto/on off is turned on in general
        # because could still be on for that day but disabled
        if self.coordinator.lm.current_status.get("global_auto") != "Enabled":
            return None

        # parse the schedule for the day
        schedule_day = self.coordinator.lm.schedule[date.weekday()]
        if schedule_day["enable"] == "Disabled":
            return None
        hour_on, minute_on = schedule_day["on"].split(":")
        hour_off, minute_off = schedule_day["off"].split(":")
        return CalendarEvent(
            start=date.replace(
                hour=int(hour_on),
                minute=int(minute_on),
            ),
            end=date.replace(
                hour=int(hour_off),
                minute=int(minute_off),
            ),
            summary=f"Machine {self.coordinator.lm.true_model_name} ({self.coordinator.lm.serial_number}) on",
            description="Machine is scheduled to turn on at the start time and off at the end time",
        )
