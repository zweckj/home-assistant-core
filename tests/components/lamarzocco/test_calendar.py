"""Tests for La Marzocco calendar."""

from copy import deepcopy
from datetime import datetime, timedelta
from unittest.mock import MagicMock

from freezegun.api import FrozenDateTimeFactory
from lmcloud.const import WeekDay
from lmcloud.exceptions import RequestNotSuccessful
import pytest
from syrupy import SnapshotAssertion

from homeassistant.components.calendar import (
    DOMAIN as CALENDAR_DOMAIN,
    EVENT_END_DATETIME,
    EVENT_START_DATETIME,
    SERVICE_GET_EVENTS,
)
from homeassistant.components.lamarzocco.calendar import (
    ATTR_DAY_OF_WEEK,
    ATTR_ENABLE,
    ATTR_TIME_OFF,
    ATTR_TIME_ON,
    SERVICE_AUTO_ON_OFF_ENABLE,
    SERVICE_AUTO_ON_OFF_TIMES,
)
from homeassistant.components.lamarzocco.const import DOMAIN
from homeassistant.const import ATTR_ENTITY_ID
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import HomeAssistantError
from homeassistant.helpers import entity_registry as er
from homeassistant.util import dt as dt_util

from . import async_init_integration

from tests.common import MockConfigEntry


async def test_calendar_events(
    hass: HomeAssistant,
    mock_lamarzocco: MagicMock,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
    freezer: FrozenDateTimeFactory,
) -> None:
    """Test the calendar."""

    test_time = datetime(2024, 1, 12, 11, tzinfo=dt_util.DEFAULT_TIME_ZONE)
    freezer.move_to(test_time)

    await async_init_integration(hass, mock_config_entry)

    serial_number = mock_lamarzocco.serial_number

    state = hass.states.get(f"calendar.{serial_number}_auto_on_off_schedule")
    assert state
    assert state == snapshot

    entry = entity_registry.async_get(state.entity_id)
    assert entry
    assert entry == snapshot

    events = await hass.services.async_call(
        CALENDAR_DOMAIN,
        SERVICE_GET_EVENTS,
        {
            ATTR_ENTITY_ID: f"calendar.{serial_number}_auto_on_off_schedule",
            EVENT_START_DATETIME: test_time,
            EVENT_END_DATETIME: test_time + timedelta(days=23),
        },
        blocking=True,
        return_response=True,
    )

    assert events == snapshot


async def test_no_calendar_events_global_disable(
    hass: HomeAssistant,
    mock_lamarzocco: MagicMock,
    mock_config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
    freezer: FrozenDateTimeFactory,
) -> None:
    """Assert no events when global auto on/off is disabled."""

    mock_lamarzocco.config.auto_on_off_schedule.enable = False
    test_time = datetime(2024, 1, 12, 11, tzinfo=dt_util.DEFAULT_TIME_ZONE)
    freezer.move_to(test_time)

    await async_init_integration(hass, mock_config_entry)

    serial_number = mock_lamarzocco.serial_number

    state = hass.states.get(f"calendar.{serial_number}_auto_on_off_schedule")
    assert state

    events = await hass.services.async_call(
        CALENDAR_DOMAIN,
        SERVICE_GET_EVENTS,
        {
            ATTR_ENTITY_ID: f"calendar.{serial_number}_auto_on_off_schedule",
            EVENT_START_DATETIME: test_time,
            EVENT_END_DATETIME: test_time + timedelta(days=23),
        },
        blocking=True,
        return_response=True,
    )
    assert events == snapshot


# test services


async def test_service_auto_on_off_enable(
    hass: HomeAssistant,
    mock_lamarzocco: MagicMock,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test the La Marzocco auto on/off enable service."""

    await async_init_integration(hass, mock_config_entry)

    await hass.services.async_call(
        DOMAIN,
        SERVICE_AUTO_ON_OFF_ENABLE,
        {
            ATTR_ENTITY_ID: f"calendar.{mock_lamarzocco.serial_number}_auto_on_off_schedule",
            ATTR_DAY_OF_WEEK: "monday",
            ATTR_ENABLE: False,
        },
        blocking=True,
    )

    schedule = deepcopy(mock_lamarzocco.config.auto_on_off_schedule)
    schedule.days[WeekDay.MONDAY].enabled = False
    assert len(mock_lamarzocco.set_schedule.mock_calls) == 1
    mock_lamarzocco.set_schedule.assert_called_once_with(schedule=schedule)


async def test_service_set_auto_on_off_times(
    hass: HomeAssistant,
    mock_lamarzocco: MagicMock,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test the La Marzocco auto on/off times service."""

    await async_init_integration(hass, mock_config_entry)

    await hass.services.async_call(
        DOMAIN,
        SERVICE_AUTO_ON_OFF_TIMES,
        {
            ATTR_ENTITY_ID: f"calendar.{mock_lamarzocco.serial_number}_auto_on_off_schedule",
            ATTR_ENABLE: True,
            ATTR_DAY_OF_WEEK: "tuesday",
            ATTR_TIME_ON: "08:30:00",
            ATTR_TIME_OFF: "17:00:00",
        },
        blocking=True,
    )

    assert len(mock_lamarzocco.set_schedule.mock_calls) == 1
    mock_lamarzocco.set_schedule.assert_called_once_with(
        schedule=mock_lamarzocco.config.auto_on_off_schedule
    )

    assert len(mock_lamarzocco.set_schedule_day.mock_calls) == 1
    mock_lamarzocco.set_schedule_day.assert_called_once_with(
        day=WeekDay.TUESDAY,
        enabled=True,
        h_on=8,
        m_on=30,
        h_off=17,
        m_off=0,
    )


async def test_service_call_error(
    hass: HomeAssistant,
    mock_lamarzocco: MagicMock,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test an exception during the service call."""

    mock_lamarzocco.set_schedule.side_effect = RequestNotSuccessful("BadRequest")

    await async_init_integration(hass, mock_config_entry)

    with pytest.raises(
        HomeAssistantError, match="Service call encountered an error: BadRequest"
    ):
        await hass.services.async_call(
            DOMAIN,
            SERVICE_AUTO_ON_OFF_ENABLE,
            {
                ATTR_ENTITY_ID: f"calendar.{mock_lamarzocco.serial_number}_auto_on_off_schedule",
                ATTR_DAY_OF_WEEK: "monday",
                ATTR_ENABLE: True,
            },
            blocking=True,
        )
