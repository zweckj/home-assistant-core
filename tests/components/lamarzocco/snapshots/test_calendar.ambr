# serializer version: 1
# name: test_calendar_events
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'all_day': False,
      'description': 'Machine is scheduled to turn on at the start time and off at the end time',
      'end_time': '2024-01-12 16:00:00',
      'friendly_name': 'GS01234 Auto on/off schedule',
      'location': '',
      'message': 'Machine GS3 AV (GS01234) on',
      'start_time': '2024-01-12 06:00:00',
    }),
    'context': <ANY>,
    'entity_id': 'calendar.gs01234_auto_on_off_schedule',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_calendar_events.1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'calendar',
    'entity_category': None,
    'entity_id': 'calendar.gs01234_auto_on_off_schedule',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Auto on/off schedule',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'auto_on_off_schedule',
    'unique_id': 'GS01234_auto_on_off_schedule',
    'unit_of_measurement': None,
  })
# ---
# name: test_calendar_events.2
  dict({
    'calendar.gs01234_auto_on_off_schedule': dict({
      'events': list([
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-12T16:00:00-08:00',
          'start': '2024-01-12T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-13T16:00:00-08:00',
          'start': '2024-01-13T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-14T16:00:00-08:00',
          'start': '2024-01-14T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-15T16:00:00-08:00',
          'start': '2024-01-15T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-16T16:00:00-08:00',
          'start': '2024-01-16T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-17T16:00:00-08:00',
          'start': '2024-01-17T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-18T16:00:00-08:00',
          'start': '2024-01-18T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-19T16:00:00-08:00',
          'start': '2024-01-19T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-20T16:00:00-08:00',
          'start': '2024-01-20T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-21T16:00:00-08:00',
          'start': '2024-01-21T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-22T16:00:00-08:00',
          'start': '2024-01-22T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-23T16:00:00-08:00',
          'start': '2024-01-23T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-24T16:00:00-08:00',
          'start': '2024-01-24T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-25T16:00:00-08:00',
          'start': '2024-01-25T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-26T16:00:00-08:00',
          'start': '2024-01-26T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-27T16:00:00-08:00',
          'start': '2024-01-27T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-28T16:00:00-08:00',
          'start': '2024-01-28T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-29T16:00:00-08:00',
          'start': '2024-01-29T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-30T16:00:00-08:00',
          'start': '2024-01-30T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-31T16:00:00-08:00',
          'start': '2024-01-31T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-02-01T16:00:00-08:00',
          'start': '2024-02-01T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-02-02T16:00:00-08:00',
          'start': '2024-02-02T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-02-03T16:00:00-08:00',
          'start': '2024-02-03T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-02-04T16:00:00-08:00',
          'start': '2024-02-04T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
      ]),
    }),
  })
# ---
# name: test_no_calendar_events_global_disable
  dict({
    'calendar.gs01234_auto_on_off_schedule': dict({
      'events': list([
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-12T16:00:00-08:00',
          'start': '2024-01-12T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-13T16:00:00-08:00',
          'start': '2024-01-13T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-14T16:00:00-08:00',
          'start': '2024-01-14T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-15T16:00:00-08:00',
          'start': '2024-01-15T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-16T16:00:00-08:00',
          'start': '2024-01-16T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-17T16:00:00-08:00',
          'start': '2024-01-17T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-18T16:00:00-08:00',
          'start': '2024-01-18T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-19T16:00:00-08:00',
          'start': '2024-01-19T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-20T16:00:00-08:00',
          'start': '2024-01-20T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-21T16:00:00-08:00',
          'start': '2024-01-21T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-22T16:00:00-08:00',
          'start': '2024-01-22T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-23T16:00:00-08:00',
          'start': '2024-01-23T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-24T16:00:00-08:00',
          'start': '2024-01-24T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-25T16:00:00-08:00',
          'start': '2024-01-25T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-26T16:00:00-08:00',
          'start': '2024-01-26T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-27T16:00:00-08:00',
          'start': '2024-01-27T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-28T16:00:00-08:00',
          'start': '2024-01-28T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-29T16:00:00-08:00',
          'start': '2024-01-29T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-30T16:00:00-08:00',
          'start': '2024-01-30T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-01-31T16:00:00-08:00',
          'start': '2024-01-31T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-02-01T16:00:00-08:00',
          'start': '2024-02-01T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-02-02T16:00:00-08:00',
          'start': '2024-02-02T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-02-03T16:00:00-08:00',
          'start': '2024-02-03T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
        dict({
          'description': 'Machine is scheduled to turn on at the start time and off at the end time',
          'end': '2024-02-04T16:00:00-08:00',
          'start': '2024-02-04T06:00:00-08:00',
          'summary': 'Machine GS3 AV (GS01234) on',
        }),
      ]),
    }),
  })
# ---
